library/std/src/sys/pal/uefi/os.rs

```diff
@@ -171,44 +172,70 @@ pub fn current_exe() -> io::Result<PathBuf> {
     helpers::device_path_to_text(protocol).map(PathBuf::from)
 }
 
-pub struct Env(!);
+#[derive(Clone)]
+pub struct Env {
+    last_var_name: Vec<u16>,
+    last_var_guid: r_efi::efi::Guid,
```

Comment:
This needs to cache all environment variables, as `Env` is ["[a]n iterator over a snapshot of the environment variables"](https://doc.rust-lang.org/nightly/std/env/struct.Vars.html).